File: 011 Intersection Sort/Intersection Sort.js
function insertionSort(array) {
  for (let i = 0; i < array.length; i++) {
		let j = i;
		while (j > 0 && array[j] < array[j - 1]){
			swap(j, j-1, array)
			j -= 1;
		}
	}
	return array;
}

function swap(a, b, array) {
	[array[a],array[b]] = [array[b],array[a]]
}

File: 011 Intersection Sort/Intersection Sort.py
def insertionSort(array):
	for i in range(len(array)):
		j = i
		while j > 0 and array[j] < array[j - 1]:
			swap(j, j - 1, array)
			j -= 1
	return array

def swap(a, b, array):
	array[a], array[b] = array[b], array[a]